packages/nvidia_nat_autogen/src/nat/plugins/autogen/tool_wrapper.py

```
@@ -0,0 +1,181 @@
+                            param_name,
+                            name,
+                        )
+
+                    default = model_field.default if model_field.is_required() is False else inspect._empty
+                    params.append(
+                        inspect.Parameter(param_name,
+                                          inspect.Parameter.POSITIONAL_OR_KEYWORD,
+                                          annotation=resolved_type,
+                                          default=default))
+                    annotations[param_name] = resolved_type
+                setattr(func_to_wrap, "__signature__", inspect.Signature(parameters=params))
```

Ruff (0.14.10)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)